concrete-benchmark/src/synthesizer.rs

```diff
@@ -0,0 +1,558 @@
+//! A module containing traits to generate dummy values for benchmarks.
+//!
+//! This module contains abstractions to create the input values needed to benchmark an engine
+//! trait. For instance, if we want to benchmark an engine trait which add two lwe ciphertexts,
+//! we first have to be able to provide those ciphertext. The benchmarked engine may not implement
+//! all the traits needed to create those ciphertexts on its own though. For this reason we need a
+//! different tool to create those input values.
+//!
+//! The [`Synthesizer`] structure contains all the necessary engines, used to generate the different
+//! types needed for benchmarking (all the types must be possible to generate with the engines
```

Comment:
sorry I don't understand the sentence inside the parenthesis :disappointed: Do you mean that we must ensure the synthesizer makes it possible to generate data for any kind of entity?